### DISCUSSION TITLE
Release Planning v00.05-beta

### DISCUSSION THREAD

**Discussion by xblax** [xblax/flashforge_ad5m_klipper_mod#149] [Development]
Let's start a separate discussion to get everything together and ready for the next v00.05-beta release.

I have integrated all the current feature branches on master. I had to revert the switch to libsvgtiny-pixbuf for klipperscreen because it was leaking memory. Seems to be working fine now, did a test print with timelapse and render.

I did also cherry-pick the G17 support in the klipper submodule.

What else should be done? I think at least including the support for the new flashforge firmware.

**Comment by KaruroChori** [comment]
Honestly, I don't see anything major missing or not working.
I never used the klipperscreen version myself, but plenty people did and the issues it had were already reported.

Maybe there is a bit of cleanup in the documentation, and it would be great to distribute orcaslicer profiles alongside the release.

**Comment by consp** [comment]
I'll try to finish guppyscreen integration but I'm not having a lot of time the coming days. Tried fix most outstanding bugs in klipperscreen last week but for the wifi issues some logs would be nice since I have no clue why they are failing at this point. One bug I know of was an "unknown channel" in use, someone was using one of the higher 5.8ghz channels which are not normally used, which caused a lookup error in klipperscreen. That one should be fixed.

**Reply by consp** [reply]
Looks like most things work, guppyscreen wpa also works now (for me), at least I can reconnect again without too much issues.

@xblax I merged the guppyscreen main branch into "our" version to keep it up to date, so if you rebuild you must do a clean one so it fetches the latest git commit.

**Reply by xblax** [reply]
Ok. I'm doing a clean build for the release anyway. But maybe we should reference a fixed revision in the future so that changes are also visible here in the repo.

**Reply by consp** [reply]
My plan is to make patch files which can be input into the main branch, but for now I'm sticking to this as it's easier for development.

**Comment by KaruroChori** [comment]
I opened the branch https://github.com/xblax/flashforge_ad5m_klipper_mod/tree/docs/0.5b to keep track of the changes in the documentation for the new version.

**Comment by xblax** [comment]
Just FYI, I'm on vacation this week, so my time is a bit limited and have not printer to test while traveling. Will try to prepare the new release next week.

@consp 
If we don't have Guppyscreen in a state we feel it can be published we can just omit it from the release then, and make another release later on or label it as alpha for now. Free time is always limited, it's done when it's done :)

**Reply by consp** [reply]
> Guppyscreen in a state we feel it can be published we can just omit it from the release then

I'm currently mainly using it so for "don't press the screen" mode it works fine :)

We shall see, probably have some time during the weekend to spend on finetuning it a bit. Mainly worried about the wpa/iwl "cooperation" at this point. You mentioned networkmanager before, but I do not think we can run that with mdev, but if it could that would be a way around all the wifi sublayers of obfuscation and a possible way to get both killerscreen and guppyscreen to cooperate with the wifi system (klipperscreen has been updated to use nm).

**Comment by xblax** [comment]
Added some more changes for v00.05-beta on branch https://github.com/xblax/flashforge_ad5m_klipper_mod/tree/feature/next

Includes update to latest Mainsail and Fluidd versions. Caveat with that is that the browser cache must be cleared for the update to be effective.

**Reply by consp** [reply]
> Caveat with that is that the browser cache must be cleared for the update to be effective.

How does fluidd/mainsail normally handle this? Or doesn't it do that? (e.g. with Cache-Control header?)

**Reply by xblax** [reply]
They set HTTP header in nginx to turn off caching, but that's not possible with busybox httpd.
````
    location = /index.html {
        add_header Cache-Control "no-store, no-cache, must-revalidate";
    }
````
Maybe we could just patch index.html to set meta tags for cache control or we just ignore it ... for now it's already cached anyways.

**Reply by consp** [reply]
Ah, makes sense it not working then. I guess a patch to index.html should be sufficient if it becomes a problem like you said.

**Reply by xblax** [reply]
Was not able to reproduce the caching issues. In all my testing Firefox and Chromium seemed to query for changes reliably. Not sure why It didn't work for me initially. We'll see when/if we get first reports.

**Comment by MrSolidus89** [comment]
are you releasing the new update today?  will wifi from display work?  Thank you.

**Reply by consp** [reply]
> will wifi from display work?

If you have issues with that, see below (please send the `/var/log/KlipperScreen.log` file).

**Comment by consp** [comment]
@xblax could you add the folowing to the release notes?

* If you experience wifi issues or have no wifi stations when using Klipperscreen, please save the `/var/log/KlipperScreen.log` file and post it as an issue if possible.

**Reply by xblax** [reply]
I have included it. The draft release is already available, the build is running now. If I forgot something in the notes you can also add it directly.
